### PR TITLE
Allow constants longer than one character

### PR DESCRIPTION
**Pull request by evouga**

### CODE REVIEW

**Review by ghamerly — APPROVED** [review]
Looks good to me.

**Review by Tagl — COMMENTED** [review]
Should names be allowed to start with a digit, dash, or period?

**Comment by niemela** [comment]
> Should names be allowed to start with a digit, dash, or period?

...or consist only of digits? I think no. 

Maybe we want the same as for filenames?: `[a-zA-Z0-9][a-zA-Z0-9_.-]*[a-zA-Z0-9]`

**Comment by Tagl** [comment]
That would allow digits at start, I think: `[a-zA-Z_][a-zA-Z0-9_]*` would be more appropriate.

**Comment by niemela** [comment]
> That would allow digits at start, 

Good point. I just copied what we have for files. Should *that* change? It feels a little bit strange that the requirements are so different?

> I think: `[a-zA-Z_][a-zA-Z0-9_]*` would be more appropriate.

That does sound better.

**Comment by evouga** [comment]
No, I think it’s ok for file names to start with numbers since this is super useful for ordering test cases.

File: spec/2023-07-draft.md [hunk 1]
```diff
@@ -302,7 +302,7 @@ File endings in parenthesis are not used for determining language.
 ### Constants
 
 A map of names to values.
-Names must match the following regex: `[a-zA-Z0-9_.-]`.
+Names must match the following regex: `[a-zA-Z0-9_.-]+`.
```
Comment (niemela) [inline spec/2023-07-draft.md:305]:
```suggestion
Names must match the following regex: `[a-zA-Z_][a-zA-Z0-9_]*`.
```

**Review by niemela — CHANGES_REQUESTED** [review]

**Review by niemela — APPROVED** [review]

**Pull request merged**